api/apiclient.go

```diff
@@ -537,11 +540,15 @@ func newWebsocketDialer(cfg *websocket.Config, opts DialOpts) func(<-chan struct
 			if err == nil {
 				return conn, nil
 			}
-			if !a.HasNext() || isX509Error(err) {
-				// We won't reconnect when there's an X509 error
-				// because we're not going to succeed if we retry
-				// in that case.
-				logger.Infof("error dialing %q: %v", cfg.Location, err)
+			if isX509Error(err) {
+				// We won't reconnect when there's an X509
+				// error because we're not going to succeed if
+				// we retry in that case.
+				logger.Errorf("certificate error dialing %q: %v", cfg.Location, err)
```

Comment:
It is, aye. It's something I added for clarity in my own testing and thought was worth keeping.